pkg/auditlog/auditlog.go

```diff
@@ -0,0 +1,88 @@
+// Package alog is a package to send audit logs events to WorkOS.
+//
+// Example:
+//   func main() {
+//       alog.SetAPIKey("my_api_key")
+//       defer alog.Close()
+//
+//       // Wherever you need to publish an audit log event:
+//       alog.Publish(alog.Event{
+//           Action:     "document.viewed",
+//           ActionType: "r",
+//           ActorName:  "Jairo Kunde",
+//           ActorID:    "user_01DGZ0FAXN978HCET66Q98QMTQ",
+//           Group:      "abstract.com",
+//           Location:   "55.27.223.26",
+//           OccurredAt: time.Now(),
+//           TargetName: "central.class",
+//           TargetID:   "doc_01DGZ0FAXP4HA4X0BVFKS0ZH4Y",
+//       })
+//   }
+package alog
+
+import (
+	"os"
+	"time"
+)
+
+var (
+	// DefaultPublisher is the publisher used by SetAPIKey, Publish and Close
+	// functions.
+	DefaultPublisher = &Publisher{
+		APIKey:    os.Getenv("WORKOS_API_KEY"),
+		Endpoint:  "https://api.workos.com/events",
+		QueueSize: 512,
+	}
+
+	currentLocation string
+)
+
+func init() {
+	currentLocation, _ = os.Hostname()
+}
+
+// SetAPIKey sets the WorkOS API key to use when using Publish.
+func SetAPIKey(k string) {
+	DefaultPublisher.APIKey = k
+}
+
+// Publish publishes the given events.
+func Publish(events ...Event) {
+	DefaultPublisher.Publish(events...)
+}
+
+// Close stops publishings audit log events and releases allocated resources.
+// It waits for pending events to be sent before returning.
+func Close() {
+	DefaultPublisher.Close()
+}
+
+// Event represents an Audit Log event.
+type Event struct {
+	Action     string                 `json:"action"`
```

Comment:
There is an `Action` type:

```suggestion
	Action     Action                 `json:"action"`
```

Reply:
looks like the `Action` type in the current implementation provide methods to extract `Category`, `Name` and `Environment` from a string.

I was not able to find where those method are used in the implementation except in the `action_test.go`.

What would be the scenario where a user would use the type?

Reply:
When this was implemented the API allowed you to prefix events with an environment as the first part of an action name separated by a dot. Working with action names I often try to avoid splitting the dots as much as I can, so this type encapsulates the logic so it doesn't need to be written every time you're trying to get the first part of the action name or the last part.

Reply:
What I mean is that the sdk only provide create operation. Once an event is created, what is the scenario where I would need to parse the event string?

Also do we need a type for this?
I think a function with the following signature:
```go
func ParseEvent(event string) (category, name, env string) 
```
would be more adapted to the need of getting the different part of the event.

The difference come in term of usage, each time we want to fill an event, we have to wrap the value like `auditlog.Event("my.event.org")`.
Its pretty verbose just to handle the scenario where we would want to deconstruct it after publish.

Reply:
The library only provides the create API now, but it will eventually provide a read API and query API. By also requiring a custom type it gives us the ability to call functions on it like `action.IsValid()` to ensure it is valid.

@maxchehab are there any validations on action names?

Reply:
@dewski I updated my previous comment, I was referring to action.

Reply:
Does the action format under the form of `category.action.env` is a good practice or it have to be enforced?

Reply:
> are there any validations on action names?
@dewski 

Nope, any names are acceptable.